src/core/providers/qgsprovidersublayermodel.cpp

```diff
@@ -656,13 +656,17 @@ bool QgsProviderSublayerProxyModel::filterAcceptsRow( int source_row, const QMod
   if ( !mIncludeSystemTables && static_cast< Qgis::SublayerFlags >( sourceModel()->data( sourceIndex, static_cast< int >( QgsProviderSublayerModel::Role::Flags ) ).toInt() ) & Qgis::SublayerFlag::SystemTable )
     return false;
 
+  if ( !mIncludeEmptyLayers && sourceModel()->data( sourceIndex, static_cast< int >( QgsProviderSublayerModel::Role::FeatureCount ) ) == 0 )
+    return false;
+
   if ( mFilterString.trimmed().isEmpty() )
     return true;
 
   if ( sourceModel()->data( sourceIndex, static_cast< int >( QgsProviderSublayerModel::Role::Name ) ).toString().contains( mFilterString, Qt::CaseInsensitive ) )
     return true;
 
-  if ( sourceModel()->data( sourceIndex, static_cast< int >( QgsProviderSublayerModel::Role::Description ) ).toString().contains( mFilterString, Qt::CaseInsensitive ) )
+  // check against the Description column's edit role so that Feature count is searchable
+  if ( sourceModel()->data( sourceModel()->index( source_row, 1, source_parent ), static_cast< int >( Qt::EditRole ) ).toString().contains( mFilterString, Qt::CaseInsensitive ) )
```

Comment:
why edit role here? (In any case it'd be cleaner to just check against the FeatureCount role separately)

Reply:
My rationale was to *search exactly what you see*, in case more info was concatenated into Description at some point.

Reply:
I meant why not the DisplayRole? EditRole is a strange choice for a non-editable model.

Reply:
Doh! I got this wrong the whole time! :)